coreos-iso-embed-ignition

```diff
@@ -0,0 +1,178 @@
+    return offset, length
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Embed an Ignition config in a CoreOS live ISO image.")
+    subparsers = parser.add_subparsers(dest='command', title='subcommands',
+            required=True,
+            description='Use <subcommand> -h for more info.')
+
+    embed = subparsers.add_parser('embed',
+            description='Embed an Ignition config in an ISO image.',
+            help='Embed an Ignition config in an ISO image')
+    embed.add_argument('-c', '--config', metavar='PATH', default='-',
+            type=argparse.FileType('r'),
+            help='Ignition config to embed [default: stdin]')
+    embed.add_argument('-f', '--force', action='store_true',
+            help='Overwrite an existing embedded Ignition config')
+    embed.add_argument('-o', '--output', metavar='PATH',
+            type=argparse.FileType('x+b'),
+            help='Copy to a new file, instead of modifying in place')
+    embed.set_defaults(command=do_embed)
+
+    show = subparsers.add_parser('show',
+            description='Show the embedded Ignition config from an ISO image.',
+            help='Show the embedded Ignition config from an ISO image')
+    show.set_defaults(command=do_show)
+
+    remove = subparsers.add_parser('remove',
+            description='Remove an existing embedded Ignition config from an ISO image.',
+            help='Remove an existing embedded Ignition config')
+    remove.add_argument('-o', '--output', metavar='PATH',
+            type=argparse.FileType('x+b'),
+            help='Copy to a new file, instead of modifying in place')
+    remove.set_defaults(command=do_remove)
+
+    for subp in embed, show, remove:
+        subp.add_argument('image', metavar='ISO',
+                type=argparse.FileType('r+b'),
+                help='ISO image')
+
+    return parser.parse_args()
+
+
+def do_embed(args, fh, offset, length):
+    cpio = make_cpio(args.config)
+    if len(cpio) > length:
+        raise EmbedException('Compressed Ignition config is too large: {} > {}'.format(len(cpio), length))
+    if not args.force:
+        fh.seek(offset)
+        if fh.read(length) != bytes(length):
+            raise EmbedException('This ISO image already has an embedded Ignition config; use -f to force.')
+    fh.seek(offset)
+    fh.write(cpio)
+    fh.write(bytes(length - len(cpio)))
+
+
+def do_show(args, fh, offset, length):
+    fh.seek(offset)
+    buf = fh.read(length)
+    if len(buf) < length:
+        raise EmbedException("Couldn't read embedded Ignition config.")
+    if buf == bytes(length):
+        raise EmbedException("No embedded Ignition config.")
+    config = extract_cpio(buf)
+    print(config.decode(), end='')
+
+
+def do_remove(args, fh, offset, length):
+    fh.seek(offset)
+    fh.write(bytes(length))
+
+
+def main():
+    args = parse_args()
+    image = CopiedFileHolder(args.image)
+    if getattr(args, 'output', None) is not None:
```

Comment:
could we just set the default value for the output arg to be None and just say `if args.output:` ?

Reply:
In this case, the issue is that some subcommands have an `output` argument and some don't.

Reply:
+1 - I missed that point. Looks good